modules/createTransitionManager.js

```diff
@@ -101,10 +101,9 @@ export default function createTransitionManager(history, routes) {
   const RouteHooks = Object.create(null)
 
   function getRouteHooksForRoutes(routes) {
-    return routes.reduce(function (hooks, route) {
-      hooks.push.apply(hooks, RouteHooks[getRouteID(route)])
-      return hooks
-    }, [])
+    return routes
+      .map(route => RouteHooks[getRouteID(route)])
```

Comment:
Oops, this should probably be `getRouteID(route, false)`, actually. No reason to create route IDs here. The `undefined` will safely avoid colliding with any actual keys under `RouteHooks`.